src/lib.rs

```diff
@@ -98,6 +88,19 @@
 //! At this time, we do not recommend use for security-critical applications.
 //! Use at your own risk.
 //!
+//! # Useful features
+//!
+//! The main interfaces allow a [`Participant`] to process messages from
+//! other participants, producing a set of outgoing messages and, if the
+//! protocol completed, an output.
+//!
+//! The library includes logic to handle messages received outside the strict
```

Comment:
This makes sense to me since I already understand how all this works. But for a new user this might be a big confusing.

I would add something like "Messages may arrive from the network before a Participant is ready to process it. The library includes..."

I would change the wording from "cached" to "stored"